app/controllers/courses_controller.rb

```diff
@@ -171,10 +171,24 @@ def sync_roster
     if params[:include_instructors] == 'true'
       roles.append(LtiDeployment::LTI_ROLES[:instructor])
     end
-    @current_job = LtiRosterSyncJob.perform_later(deployment, @current_course,
-                                                  roles,
-                                                  can_create_users: allowed_to?(:lti_manage?, with: UserPolicy),
-                                                  can_create_roles: allowed_to?(:manage?, with: RolePolicy))
+    job_args = {}
+    job_args[:deployment] = deployment.id
+    job_args[:role_types] = roles
+    job_args[:can_create_users] = allowed_to?(:lti_manage?, with: UserPolicy)
+    job_args[:can_create_roles] = allowed_to?(:manage?, with: RolePolicy)
+    name = "#{@current_course.name}_#{root_path.tr!('/', '')}"
+    if params[:automatic_sync] == 'true'
+      config = {}
+      config[:class] = 'ActiveJob::QueueAdapters::ResqueAdapter::JobWrapper'
+      config[:args] = { job_class: 'LtiRosterSyncJob', arguments: [job_args] }
+      config[:cron] = Settings.lti.sync_schedule
+      config[:persist] = true
+      config[:queue] = 'DEFAULT_QUEUE'
```

Comment:
This doesn't seem like it should be hard-coded. Can we use the `LtiRosterSyncJob.queue_name` method?